internal/configs/bootnode/bootnode.go

```diff
@@ -0,0 +1,316 @@
+	// enable p2p NAT. NAT Manager takes care of setting NAT port mappings, and discovering external addresses
+	NAT bool
+	// custom user agent; explicitly set the user-agent, so we can differentiate from other Go libp2p users
+	UserAgent string
+	// p2p dial timeout
+	DialTimeout time.Duration
+	// P2P multiplexer type, should be comma separated (mplex, Yamux)
+	Muxer string
+	// No relay services, direct connections between peers only
+	NoRelay bool
+}
+
+type GeneralConfig struct {
+	NodeType               string
+	NoStaking              bool
+	ShardID                int
+	IsArchival             bool
+	IsBackup               bool
+	IsBeaconArchival       bool
+	IsOffline              bool
+	DataDir                string
+	TraceEnable            bool
+	EnablePruneBeaconChain bool
+	RunElasticMode         bool
+}
+
+type TiKVConfig struct {
+	Debug bool
+
+	PDAddr                      []string
+	Role                        string
+	StateDBCacheSizeInMB        uint32
+	StateDBCachePersistencePath string
+	StateDBRedisServerAddr      []string
+	StateDBRedisLRUTimeInDay    uint32
+}
+
+type ShardDataConfig struct {
+	EnableShardData bool
+	DiskCount       int
+	ShardCount      int
+	CacheTime       int
+	CacheSize       int
+}
+
+type GasPriceOracleConfig struct {
+	// the number of blocks to sample
+	Blocks int
+	// the number of transactions to sample, per block
+	Transactions int
+	// the percentile to pick from there
+	Percentile int
+	// the default gas price, if the above data is not available
+	DefaultPrice int64
+	// the maximum suggested gas price
+	MaxPrice int64
+	// when block usage (gas) for last `Blocks` blocks is below `LowUsageThreshold`,
+	// we return the Default price
+	LowUsageThreshold int
+	// hack: our block header reports an 80m gas limit, but it is actually 30M.
+	// if set to non-zero, this is applied UNCHECKED
+	BlockGasLimit int
+}
+
+type PprofConfig struct {
+	Enabled            bool
+	ListenAddr         string
+	Folder             string
+	ProfileNames       []string
+	ProfileIntervals   []int
+	ProfileDebugValues []int
+}
+
+type LogConfig struct {
+	Console       bool
+	Folder        string
+	FileName      string
+	RotateSize    int
+	RotateCount   int
+	RotateMaxAge  int
+	Verbosity     int
+	VerbosePrints LogVerbosePrints
+	Context       *LogContext `toml:",omitempty"`
+}
+
+type LogVerbosePrints struct {
+	Config bool
+}
+
+func FlagSliceToLogVerbosePrints(verbosePrintsFlagSlice []string) LogVerbosePrints {
+	verbosePrints := LogVerbosePrints{}
+	verbosePrintsReflect := reflect.Indirect(reflect.ValueOf(&verbosePrints))
+	for _, verbosePrint := range verbosePrintsFlagSlice {
+		verbosePrint = strings.Title(verbosePrint)
+		field := verbosePrintsReflect.FieldByName(verbosePrint)
+		if field.IsValid() && field.CanSet() {
+			field.SetBool(true)
+		}
+	}
+
+	return verbosePrints
+}
+
+type LogContext struct {
+	IP   string
+	Port int
+}
+
+type SysConfig struct {
+	NtpServer string
+}
+
+type HttpConfig struct {
```

Comment:
there is no rosetta for bootnode right ?

Reply:
removed